### PR TITLE
Relation styling update

### PR DESCRIPTION
**Pull request by anthonydillon**

### CODE REVIEW

**Comment by anthonydillon** [comment]
## QA
- Add two relatable charms to the canvas and create a relation
- Check the relation line match the design [1]
- Commit the charms and relation and see the relation line colour updates with the relation connectors
- Add a bunch of units and deploy them
- Hit Ctrl+S to simulate and wait for the relation to go into error state and check it all looks correct

[1] https://drive.google.com/a/canonical.com/file/d/0B2sqJRO7vy3ZVVBIWldXbkFGYWs/view?usp=sharing

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3001/
Test PASSed.

File: jujugui/static/gui/src/app/views/topology/relation.js [hunk 1]
```diff
@@ -359,6 +359,26 @@ YUI.add('juju-topology-relation', function(Y) {
             // Style relation lines differently depending on status.
             return 'relation ' + d.aggregatedStatus;
           });
+      enter.append('circle')
```
Comment (hatched) [inline jujugui/static/gui/src/app/views/topology/relation.js]: I'm assuming that these are the little knobs on the end of the lines? Can you add a comment to this effect

File: jujugui/static/gui/src/app/views/topology/relation.js [hunk 2]
```diff
@@ -424,6 +447,28 @@ YUI.add('juju-topology-relation', function(Y) {
             // Style relation lines differently depending on status.
             return 'relation ' + relation.aggregatedStatus;
           });
+
+      var deltaX = s[0] - t[0];
```
Comment (hatched) [inline jujugui/static/gui/src/app/views/topology/relation.js]: Can you add some comments here as to what this is all for?

**Comment by hatched** [comment]
:+1: QA OK

Are we going to be converting those SVG's into minified sprites at some point? They look like quite complex SVG's for what they represent.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3011/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3012/
Test FAILed.

**Comment by anthonydillon** [comment]
Thanks @hatched, somehow my comments have caused the tests to fail lol. Will work on it. Regarding the SVG sprites I don't think these elements should be sprites as we then loss the ability to modify them through the CSS. To be honest, I am against spriting in general but that's for another day.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3014/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3015/
Test PASSed.

**Comment by makyo** [comment]
Awesome :+1:

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3019/
Test FAILed.

**Comment by anthonydillon** [comment]
Tests passed when running manually: http://ci.jujugui.org:8080/job/juju-gui/3023/console

**Comment by anthonydillon** [comment]
Thanks guys :shipit:

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**